src/PalindromeApp.java

```diff
@@ -0,0 +1,37 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PalindromeApp {
+    public static void main(String[] args) {
+        System.out.println(isPalindrome("Madam, I'm Adam!"));
+        System.out.println(isPalindrome("A1ll1a!"));
+        System.out.println(isPalindrome("aad,dffk1k"));
+    }
+
+    public static boolean isPalindrome(String text) {
+        Pattern p = Pattern.compile("[a-zA-Z0-9]");
```

Comment:
simplify. for example you can change it:
        Pattern p = Pattern.compile("[a-zA-Z0-9]");Matcher m = p.matcher(text);
        StringBuilder textWithoutSymbols = new StringBuilder();
        while (m.find()) {
            textWithoutSymbols.append(text.substring(m.start(), m.end()));
        }

to that one:
StringBuilder newString = new StringBuilder(text.replaceAll("[^a-zA-Z0-9]", ""));